lib/common/zstd_internal.h

```diff
@@ -391,34 +395,38 @@ MEM_STATIC unsigned ZSTD_countTrailingZeros(size_t val)
 #           if STATIC_BMI2
                 return _tzcnt_u64(val);
 #           else
-                unsigned long r = 0;
-                return _BitScanForward64( &r, (U64)val ) ? (unsigned)(r >> 3) : 0;
+                if (val != 0) {
+                    unsigned long r;
+                    _BitScanForward64(&r, (U64)val);
+                    return (unsigned)r;
+                } else {
+                    /* Should not reach this code path */
+                    __assume(0);
+                }
 #           endif
+#       elif defined(_MSC_VER) && !defined(_WIN64)
```

Comment:
In this case we're already in 64-bit mode, `MEM_64bits()`. We can definitely use the fallback implementation here, if for some reason `!defined(_WIN64)`.